### PR TITLE
[RNMobile] Stop saving HTML markup representation

### LINKED ISSUE #5671: Remove video block representation save
## Description

The web version removed the `save` function ([reference](https://github.com/Automattic/jetpack/pull/30081)) because now the backend builds the HTML markup based on the block attributes. This means that the block HTML representation has changed from:

```
<!-- wp:videopress/video {"title":"Planet video","description":"Video description.","id":21,"guid":"AbcDe","cacheHtml":"<cache-html>","videoRatio":56.333333333333336,"allowDownload":false,"displayEmbed":false,"rating":"PG-13","isPrivate":true,"duration":1673} -->
<figure class="wp-block-videopress-video wp-block-jetpack-videopress jetpack-videopress-player">
    <div class="jetpack-videopress-player__wrapper">
        https://videopress.com/v/AbcDe?resizeToParent=true&amp;cover=true&amp;preloadContent=metadata&amp;useAverageColor=true
    </div>
</figure>
<!-- /wp:videopress/video -->
```

To:

```
<!-- wp:videopress/video {"title":"Planet video","description":"Video description.","caption":"","loop":true,"playsinline":true,"id":21,"guid":"AbcDe","cacheHtml":"<cache-html","posterData":{},"videoRatio":56.333333333333336,"tracks":[],"allowDownload":false,"displayEmbed":false,"rating":"PG-13","isPrivate":true,"duration":1673} /-->
```

As you can see, we now only persist the block attributes.

Following the web implementation, we can also remove that function. However, we'd also need to update the upload processors to reflect this change:
- https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoPressBlockProcessor.kt
- https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoPressUploadProcessor.swift

### PR DESCRIPTION
**Pull request by fluiddot**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5671.

## Proposed changes:

* Following the changes introduced https://github.com/Automattic/jetpack/pull/30081 in the web version, we apply the same approach in the native version.
* The main change is that the block doesn't contain HTML markup representation:

**Before:**
```
<!-- wp:videopress/video {"title":"Planet video","description":"Video description.","id":21,"guid":"AbcDe","cacheHtml":"<cache-html>","videoRatio":56.333333333333336,"allowDownload":false,"displayEmbed":false,"rating":"PG-13","isPrivate":true,"duration":1673} -->
<figure class="wp-block-videopress-video wp-block-jetpack-videopress jetpack-videopress-player">
    <div class="jetpack-videopress-player__wrapper">
        https://videopress.com/v/AbcDe?resizeToParent=true&amp;cover=true&amp;preloadContent=metadata&amp;useAverageColor=true
    </div>
</figure>
<!-- /wp:videopress/video -->
```

**After:**

```
<!-- wp:videopress/video {"title":"Planet video","description":"Video description.","caption":"","loop":true,"playsinline":true,"id":21,"guid":"AbcDe","cacheHtml":"<cache-html","posterData":{},"videoRatio":56.333333333333336,"tracks":[],"allowDownload":false,"displayEmbed":false,"rating":"PG-13","isPrivate":true,"duration":1673} /-->
```

⚠️ Once this PR is merged, we'd need to update the upload processors to reflect this change:
- https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoPressBlockProcessor.kt
- https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoPressUploadProcessor.swift

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1681483252176389-slack-C04LWMHSGLC

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

**Preparation:** 
The changes introduced in https://github.com/Automattic/jetpack/pull/30081 are not yet in WPCOM, so we need to test this using sandbox:
* Sandbox a Simple site and use changes from this PR ([instructions](https://github.com/Automattic/jetpack/pull/30134#issuecomment-1511630945)).

**Steps:**
* On the browser, navigate to the Simple site and create/open a post.
* Add a VideoPress block.
* Add a video.
* Switch to the code editor and check that the block doesn't contain HTML markup.
* Save the post.
* On the app, open the post.
* Observe that the block is properly rendered and that the video can be played.
* Switch to the HTML mode and check that the block doesn't contain HTML markup.
* Remove the block and add a new VideoPress block.
* Add a video.
* Switch to the HTML mode and check that the block doesn't contain HTML markup.
* Save the post.
* One the browser, open the post.
* Observe that the block is properly rendered and that the video can be played.
* Switch to the code editor and check that the block doesn't contain HTML markup.

### CODE REVIEW

**Review by SiobhyB — DISMISSED** [review]
I went through the testing steps to sandbox a simple site and confirmed that everything worked as expected when switching between adding blocks on the web and the app. Thanks for addressing this Carlos 🙇‍♀️ 

Going ahead to approve but noting that `tools/fixup-project-versions.sh` needs to be run against this PR before it can be merged. I'd be happy to re-approve after that's done.

**Comment by fluiddot** [comment]
> Going ahead to approve but noting that `tools/fixup-project-versions.sh` needs to be run against this PR before it can be merged. I'd be happy to re-approve after that's done.

Version bumped in https://github.com/Automattic/jetpack/pull/30134/commits/d407f95a17d28d9eb406f5ec1bcf8d3ed7215294. @SiobhyB let me know if you can re-approve the PR, thanks 🙇 !

**Review by SiobhyB — APPROVED** [review]
Re-approving, thanks @fluiddot 🙌

**Pull request merged**
